runner/common/stop.go

```diff
@@ -65,7 +65,7 @@ func truncateStop(pieces []string, stop string) ([]string, bool) {
 	return result, tokenTruncated
 }
 
-func incompleteUnicode(token string) bool {
+func IncompleteUnicode(token string) bool {
```

Comment:
Not in scope of this change:

This should use [`utf8.FullRuneInString`](https://pkg.go.dev/unicode/utf8#FullRuneInString) which is significantly faster

Reply:
`FullRuneInString` checks to see if there is a full UTF8 rune at the beginning of the string, whereas this is looking at the end. As a result, if we get back a token that has a valid UTF8 rune followed by an incomplete one, the library function will return true but ours will return false.